src/main/java/org/jabref/model/groups/PopularityGroup.java

```diff
@@ -0,0 +1,146 @@
+package org.jabref.model.groups;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import javafx.beans.property.Property;
+import javafx.beans.property.StringProperty;
+
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+
+import org.h2.mvstore.MVMap;
+import org.h2.mvstore.MVStore;
+
+import static org.jabref.gui.desktop.os.NativeDesktop.getOtherDataDir;
+
+/**
+ * This group contains popular entries that have been viewed more than once.
+ * {@link StandardField#GROUPS}.
+ */
+public class PopularityGroup extends AbstractGroup {
+    private static MVStore mvStore;
+    private static List<BibEntry> data;
+    private final Property<Integer> maxEntriesProperty;
+    private final StringProperty timePeriodProperty;
+
+    public PopularityGroup(String name, GroupHierarchyType context, List<BibEntry> entries, Property<Integer> integerProperty, StringProperty textProperty) {
+        super(name, context);
+        mvStore = getMVStore();
+        data = processPopularityGroup(entries, textProperty.getValue(), integerProperty.getValue());
+        this.maxEntriesProperty = integerProperty;
+        this.timePeriodProperty = textProperty;
+    }
+
+    public boolean contains(BibEntry entry) {
+        int viewCount = getEntryViewCount(entry);
+        int maxEntries = maxEntriesProperty.getValue();
+        if (viewCount > 0) {
+            for (int i = 0; i < Math.min(maxEntries, data.size()); i++) {
+                if (data.get(i).equals(entry)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+        @Override
+    public AbstractGroup deepCopy() {
+        // Create a copy of this popularity group with the same parameters
+        return new PopularityGroup(getName(), getHierarchicalContext(), data, maxEntriesProperty, timePeriodProperty);
+    }
+
+    @Override
+    public boolean isDynamic() {
+        // This group is dynamic since it changes based on view count
+        return true;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof PopularityGroup other)) {
+            return false;
+        }
+        return Objects.equals(getName(), other.getName())
+                && Objects.equals(getHierarchicalContext(), other.getHierarchicalContext());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getName(), getHierarchicalContext());
+    }
+
+    /**
+     * Retrieves the view count for a given BibEntry.
+     *
+     * @param entry The BibEntry whose view count is to be fetched.
+     * @return The number of views for the BibEntry, or 0 if not found.
+     */
+    public static int getEntryViewCount(BibEntry entry) {
+        MVMap<String, Integer> viewCounts = mvStore.openMap("entryViewCounts");
```

Comment:
Please try to make that a class variable. It is passed from the library tab on creation.